src/CSharpMongoMigrations/Migrations/IConditionalMigration.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace CSharpMongoMigrations.Migrations
+{
+    /// <summary>
+    /// Define condition checks
+    /// </summary>
+    public interface IConditionalMigration
```

Comment:
It shouldn't be a new interface. 
It's much better to extend the existing one **IMigration** with the default interface methods. (By default the both methods should return `false`).
If you want, you always can override them in your specific Migration class.

As a result, you don't need to create so many lines of code. 
For example, how to pass such methods into the VersionedMigration and so on. There is no need to use the composition there, because all what we need will be a `IMigration` variable.

Reply:
@VladimirRybalko I was thinking to not touch the existing interface and add new one for specific functionality. I also agree with your suggestion.

I have pushed new changes. Please have a look.